inflate_p.h

```diff
@@ -211,3 +212,59 @@ static inline void window_output_flush(PREFIX3(stream) *strm) {
     state->whave += out_bytes;
     state->whave = MIN(state->whave, state->wsize);
 }
+
+/* Behave like chunkcopy, but avoid writing beyond of legal output. */
+static inline uint8_t* chunkcopy_safe(uint8_t *out, uint8_t *from, unsigned len, uint8_t *safe) {
+    unsigned safelen = (unsigned)((safe - out) + 1);
+    len = MIN(len, safelen);
+    int olap_src = ((from >= out && from < out+len) ||
+            (from + len > out && from <= out));
```

Comment:
Might be easier to read if second line starts with `from <= out && from + len > out`

Reply:
That way it starts with `from >= out` on first line and `from <= out` on second line.